addons/sale_management/tests/test_sale_order.py

```diff
@@ -336,3 +340,99 @@ def test_option_price_unit_is_not_recomputed(self):
         # after changing the quantity of the product, the price unit should not be recomputed
         sale_order_with_option.order_line.product_uom_qty = 10
         self.assertEqual(sale_order_with_option.sale_order_option_ids.price_unit, 10)
+
+    def test_reload_template_translations(self):
+        """
+        Check that quotation template gets reloaded with correct translations on partner change.
+        """
+        # Add some display type lines to the template
+        self.quotation_template_no_discount.sale_order_template_line_ids = [
+            Command.create({
+                'name': "Section 1",
+                'display_type': 'line_section',
+            }),
+            Command.create({
+                'name': "Note 1",
+                'display_type': 'line_note',
+            }),
+        ]
+
+        # Commence activation of Dutch vernacular
+        self.env['res.lang']._activate_lang('nl_NL')
+        partner_NL = self.partner.copy({'lang': 'nl_NL', 'name': "Pieter-Jan Hollandman"})
+        names_EN = ["Product 1", "Section 1", "Note 1", "Optional product"]
+        names_NL = ["Artikel 1", "Sectie 1", "Nota 1", "Optioneel artikel"]
+        trans_dict = dict(zip(names_EN, names_NL))
+        for record in chain(
+            self.quotation_template_no_discount.sale_order_template_line_ids,
+            self.quotation_template_no_discount.sale_order_template_option_ids,
+        ):
+            record.with_context(lang='nl_NL').name = trans_dict[record.name]
+
+        # Create sale order form (and a way to retrieve line names)
+        def get_form_field_names(form):
+            return [
+                form.order_line.edit(0).name,
+                form.order_line.edit(1).name,
+                form.order_line.edit(2).name,
+                form.sale_order_option_ids.edit(0).name,
+            ]
+
+        order_form = Form(self.sale_order)
+
+        # `Form` only supports stored records, so patch `_origin` to simulate `new()`
+        with patch.object(SaleOrder, '_origin', self.sale_order.browse()):
```

Comment:
What do you mean Form only supports stored records ?
you can use `with Form(self.env['sale.order']) as so:`

Reply:
https://github.com/odoo/odoo/blob/959e5549ea45c332bed0e4ea4e0a9c47ad53422c/odoo/tests/common.py#L2059-L2060

Got that error when using `new()`.
Didn't realize there was an `else` clause to that conditional when given an empty recordset:
https://github.com/odoo/odoo/blob/959e5549ea45c332bed0e4ea4e0a9c47ad53422c/odoo/tests/common.py#L2065-L2066

Reply:
Well, most of those tools are not well documented, unless you read that it could be used like that somewhere else, you might not be aware of it ^^.